test/exit-format-sol.test.ts

```diff
@@ -141,4 +142,77 @@ describe("ExitFormat (solidity)", function () {
       BigNumber.from(amount).mul(2)
     );
   });
+
+  it("Can execute a single ERC20 asset exit", async function () {
+    const [alice] = await ethers.getSigners();
+
+    // Alice gets all of the initial minting of tokens
+    let initialSupply = ethers.utils.parseEther((1000).toString());
+    let erc20Token = await (
+      await ethers.getContractFactory("TestERC20", alice)
+    ).deploy(initialSupply);
+    await erc20Token.deployed();
+
+    // Alice transfers all tokens to the TestConsumer
+    await erc20Token.connect(alice).transfer(testConsumer.address, initialSupply);    
+    expect(await erc20Token.balanceOf(alice.address)).to.equal(0);
+    expect(await erc20Token.balanceOf(testConsumer.address)).to.equal(initialSupply);
+
+    // an exit referring to the token contract
+    const singleAssetExit: SingleAssetExit = {
+      asset: erc20Token.address,
+      metadata: "0x",
+      allocations: [
+        {
+          destination: "0x000000000000000000000000" + alice.address.slice(2), // padded alice
+          amount: initialSupply.toString(),
+          allocationType: AllocationType.simple,
+          metadata: "0x",
+        },
+      ],
+    };
+
+    // Use the exit to withdraw the tokens
+    await (await testConsumer.executeSingleAssetExit(singleAssetExit)).wait();
+    expect(await erc20Token.balanceOf(alice.address)).to.equal(initialSupply);
+    expect(await erc20Token.balanceOf(testConsumer.address)).to.equal(0);
+
+  });
+
+  it("Can execute a single ERC1155 asset exit", async function () {
```

Comment:
It would be interesting for this test to include multiple token types (managed by the same ERC1155 contract) in an exit.

Reply:
Added another test which shows multiple tokens in the same collection exit at once. Works as expected